src/Microsoft.DotNet.Wpf/src/PresentationFramework/MS/Internal/AppModel/ResourceContainer.cs

```diff
@@ -243,123 +184,111 @@ protected override PackagePart GetPartCore(Uri uri)
 
         #region Private Methods
 
-        // AppDomain.AssemblyLoad event handler. Check whether the assembly's resourcemanager has
+        // AppDomain.AssemblyLoad event handler. Check whether the assembly's ResourceManager has
         // been added to the cache. If it has, we need to update the cache with the newly loaded dll.
         private void OnAssemblyLoadEventHandler(object sender, AssemblyLoadEventArgs args)
         {
             Assembly assembly = args.LoadedAssembly;
             // This is specific for designer (Sparkle) scenario: rebuild and reload dll using Load(Byte[]).
-            // We do not care about assemblies loaded into the reflection-only context or the Gaced assemblies.
+            // We do not care about assemblies loaded into the reflection-only context or the GACed assemblies.
             // For example, in Sparkle whenever a project is built all dependent assemblies will be loaded reflection only.
             // We do no care about those. Only when a assembly is loaded into the execution context, we will need to update the cache. 
-            if ((!assembly.ReflectionOnly))
+            if (!assembly.ReflectionOnly)
             {
-                AssemblyName assemblyInfo = new AssemblyName(assembly.FullName);
+                ReadOnlySpan<char> assemblyName = ReflectionUtils.GetAssemblyPartialName(assembly);
+                ReflectionUtils.GetAssemblyVersionPlusToken(assembly, out ReadOnlySpan<char> assemblyVersion, out ReadOnlySpan<char> assemblyToken);
+                int totalLength = assemblyName.Length + assemblyVersion.Length + assemblyToken.Length;
 
-                string assemblyName = assemblyInfo.Name.ToLowerInvariant();
-                string assemblyKey = string.Empty;
-                string key = assemblyName;
+                Span<char> key = totalLength <= 256 ? stackalloc char[totalLength] : new char[totalLength];
+                assemblyName.ToLowerInvariant(key);
 
                 // Check if this newly loaded assembly is in the cache. If so, update the cache.
                 // If it is not in cache, do not do anything. It will be added on demand.
-                // The key could be Name, Name + Version, Name + PublicKeyToken, or Name + Version + PublicKeyToken.  
+                // The key could be Name; Name + Version; Name + PublicKeyToken; or Name + Version + PublicKeyToken.  
```

Comment:
(Makes me wonder which language uses semicolons this way, comma was grammatically fine.)

Reply:
I'll revert this, shouldn't have been changed. My language might as well be the same as yours, you never know.